### PR TITLE
Refactoring log upload

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request
This is refactoring of our log uploading process. 
- The process that upload the logs lives with the agent instead of the task
- The task now logs to a file and to the console.
- The task log file is synchronized to the log container periodically

### CODE REVIEW

File: src/agent/onefuzz-agent/src/log_uploader.rs [hunk 1]
```diff
@@ -0,0 +1,244 @@
+#![allow(clippy::if_same_then_else)]
```
Comment (chkeita) [inline src/agent/onefuzz-agent/src/log_uploader.rs:1]: This is the new log uploader

File: src/agent/onefuzz-agent/src/log_uploader.rs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+struct SeekableFile {
+    file: Arc<Mutex<tokio::fs::File>>,
+    start_position: u64,
+    pub len: usize,
+}
+
+impl SeekableFile {
+    fn new(file_path: &Path, start_position: usize) -> Result<Self> {
+        let mut file = std::fs::File::open(file_path)?;
+        let reamining_size = file.metadata().unwrap().len() - (start_position as u64);
```
Comment (tevoinea) [inline src/agent/onefuzz-agent/src/log_uploader.rs:78]: NIT(typo): remaining

**Review by tevoinea — APPROVED** [review]

File: src/agent/onefuzz-agent/src/worker.rs [hunk 3]
```diff
@@ -212,11 +225,32 @@ impl State<Ready> {
 
         info!("IPC connection bootstrapped");
 
+        let log_path = Path::join(&self.ctx.work_dir, "task_log.txt");
```
Comment (tevoinea) [inline src/agent/onefuzz-agent/src/worker.rs:228]: Don't forget to update this after your most recent commit

Comment (chkeita) [reply src/agent/onefuzz-agent/src/worker.rs:228]: No need, this this the path to the log file of the agent. 
My last commit updated the path to the log file in the container.

**Pull request merged**